Tribler/community/tunnel/hidden_community.py

```diff
@@ -363,21 +394,29 @@ def check_created_e2e(self, messages):
     def on_created_e2e(self, messages):
         for message in messages:
             cache = self.request_cache.pop(u"e2e-request", message.payload.identifier)
-            shared_secret = self.crypto.verify_and_generate_shared_secret(cache.hop.dh_secret, message.payload.key, 
-                                                                          message.payload.auth, 
+            shared_secret = self.crypto.verify_and_generate_shared_secret(cache.hop.dh_secret,
+                                                                          message.payload.key,
+                                                                          message.payload.auth,
                                                                           cache.hop.public_key.key.pk)
             session_keys = self.crypto.generate_session_keys(shared_secret)
 
-            _, rp_info = decode(self.crypto.decrypt_str(message.payload.rp_sock_addr, session_keys[EXIT_NODE], 
+            _, rp_info = decode(self.crypto.decrypt_str(message.payload.rp_sock_addr,
+                                                        session_keys[EXIT_NODE],
                                                         session_keys[EXIT_NODE_SALT]))
-            
-            # Since it is the seeder that chose the rendezvous_point, we're essentially losing 1 hop of anonymity
-            # at the downloader end. To compensate we add an extra hop.
+
+            # Since it is the downloader that chose the rendezvous_point, we're essentially losing 1 hop of anonymity
```

Comment:
I could be wrong, but as I recall the downloader sends a create-e2e to the seeder. The seeder selects and builds a rendezvous point, and sends a created-e2e back to the download, containing the sock_addr of the rendezvous point.